### ISSUE TITLE
Returning an array with JSON representations of messages

### ISSUE THREAD

**Issue by abelardolg** [mscdex/node-imap#858]
Hi there,

I have a controller which uses a service. Inside this service, I have got a function called "receive" where I put your logic into it.
![Captura de pantalla de 2021-11-07 11-06-31](https://user-images.githubusercontent.com/22522883/140640678-7c06c0b4-f18a-449b-aba1-f27a2f30d6c1.png)

I would like to store in a data array a JSON representation of each message to be sent to my client.

Currently, stream.on("data"...) function returns a chunk that converts its data into a string (yes, I know this is a example):
![Captura de pantalla de 2021-11-07 10-58-29](https://user-images.githubusercontent.com/22522883/140640458-e0595c49-dc7c-4314-9406-354608c16540.png)

Is it possible to convert it into a Json object?
![Captura de pantalla de 2021-11-07 11-02-10](https://user-images.githubusercontent.com/22522883/140640547-5f4d05d4-4e11-4695-bedd-244c1c53e375.png)

When I use the above JSON function, the data array stores this illegible objects:
![Captura de pantalla de 2021-11-07 11-02-43](https://user-images.githubusercontent.com/22522883/140640579-8fc72bfa-5dbd-4e9b-8c60-219e36537113.png)

To sum up:
Could we have a small example how to obtain messages, convert them into a JSON object and sent to my controller, please?

Thanks in advance. Best regards.

**Comment by naglafar** [comment]
I might have got the wrong end of the stick but you want to use [mailperser](https://www.npmjs.com/package/mailparser)

```
 const fetch = connection.fetch(results, { bodies: '', markSeen: true });
  fetch.on('message', (msg) => {
    msg.on('body', async (stream) => {
      try {
        const body = await simpleParser(stream);
        await processEmail({ body });    /// do stuff 
      } catch (error) {
        logger.error(error);
      }
    });
  });

```
